src/jit/morph.cpp

```diff
@@ -15130,11 +15099,8 @@ void Compiler::fgCheckQmarkAllowedForm(GenTree* tree)
 #else // LEGACY_BACKEND
     GenTreePtr colon = tree->gtOp.gtOp2;
 
-    assert(colon->gtOp.gtOp1->gtOper == GT_CNS_INT);
-    assert(colon->gtOp.gtOp1->AsIntCon()->IconValue() == 0);
-
-    assert(colon->gtOp.gtOp2->gtOper == GT_CNS_INT);
-    assert(colon->gtOp.gtOp2->AsIntCon()->IconValue() == 1);
+    assert(colon->gtOp.gtOp1->IsZero());
+    assert(colon->gtOp.gtOp2->IsIOne());
```

Comment:
There is no way that liberalizing this check will lead to SBCG. We have many checks that operands are compatible - so we couldn't have a float 0 and an integral 1. Further, this is on legacy backend only, and we currently only allow colon to have TYP_I_IMPL operands. The point of these asserts are to ensure that the operands are the constants zero and one, NOT to ensure that the operand types have been kept consistent. So, this gets at the real point of the asserts. I actually believe quite strongly that an excess of asserts impedes rather than improves the maintainability of the code.